mmv1/third_party/terraform/services/bigquery/resource_bigquery_table_test.go

```diff
@@ -2944,6 +3031,196 @@ resource "google_bigquery_table" "test" {
 `, datasetID, tableID)
 }
 
+func testAccBigQueryTablePrimaryKey(datasetID, tableID string) string {
+	return fmt.Sprintf(`
+  resource "google_bigquery_dataset" "foo" {
+    dataset_id = "%s"
+  }
+
+  resource "google_bigquery_table" "test" {
+	  deletion_protection = false
+    table_id   = "%s"
+    dataset_id = google_bigquery_dataset.foo.dataset_id
+
+    table_constraints {
+		primary_key {
+			columns = ["id"]
+		}
+    }
+
+    schema = <<EOH
+	[
+		{
+			"name": "ts",
+			"type": "TIMESTAMP"
+		},
+		{
+			"name": "id",
+			"type": "INTEGER"
+		}
+	]
+	EOH
+	}
+  `, datasetID, tableID)
+}
+
+func testAccBigQueryTableForeignKeys(projectID, datasetID, tableID_pk, tableID_fk string) string {
+	return fmt.Sprintf(`
+  resource "google_bigquery_dataset" "foo" {
+    dataset_id = "%s"
+  }
+
+  resource "google_bigquery_table" "table_pk" {
+	deletion_protection = false
+    table_id   = "%s"
+    dataset_id = google_bigquery_dataset.foo.dataset_id
+
+    table_constraints {
+		primary_key {
+			columns = ["id"]
+		}
+    }
+
+    schema = <<EOH
+	[
+		{
+			"name": "ts",
+			"type": "TIMESTAMP"
+		},
+		{
+			"name": "id",
+			"type": "INTEGER"
+		},
+		{
+			"name": "str",
+			"type": "STRING"
+		}
+	]
+	EOH
```

Comment:
Can you please use two empty spaces for indentations? Thanks.

Reply:
Done